src/components/Calculator.js

```diff
@@ -1,39 +1,40 @@
+import { useState } from 'react';
+import buttons from '../logic/buttons';
+import calculate from '../logic/calculate';
+
 function Calculator() {
+  const [calculation, setCalculation] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  const handleClick = (e) => {
+    setCalculation(calculate(calculation, e.target.textContent));
+  };
+
+  const { total, next } = calculation;
+  const display = next || total || '0';
+
   return (
     <div className="calculator">
-      <input type="text" className="calculator-display" readOnly />
+      <input
+        type="text"
+        className="calculator-display"
+        value={display}
+        readOnly
+      />
       <div className="calculator-buttons">
-        <button type="button">AC</button>
-        <button type="button">+/-</button>
-        <button type="button">%</button>
-        <button type="button" id="orange">
-          ÷
-        </button>
-        <button type="button">7</button>
-        <button type="button">8</button>
-        <button type="button">9</button>
-        <button type="button" id="orange">
-          x
-        </button>
-        <button type="button">4</button>
-        <button type="button">5</button>
-        <button type="button">6</button>
-        <button type="button" id="orange">
-          -
-        </button>
-        <button type="button">1</button>
-        <button type="button">2</button>
-        <button type="button">3</button>
-        <button type="button" id="orange">
-          +
-        </button>
-        <button type="button" className="btn0">
-          0
-        </button>
-        <button type="button">.</button>
-        <button type="button" id="orange">
-          =
-        </button>
+        {buttons.map((button) => (
+          <button
+            type="button"
+            key={button.name}
+            className={`${button.className} ${button.color}`}
+            onClick={handleClick}
+          >
+            {button.name}
+          </button>
+        ))}
```

Comment:
- [ ] Dear @SergioPeralta22, You did great on the project, But one of the main goals for this project is to pass props between parent and child components. As part of that, we need to pass `props` to a child compone of a `Calculate.js` component. Please make sure `Props` are passed correctly to any child components of the `Calculate.js` component.

Reply:
@TadesseDev 

Why are you trying to block this project, could you please indicate which requirement I have failed?

![image](https://github.com/SergioPeralta22/math-magicians/assets/78930134/e563de66-a003-46ca-8d5a-3e513fe653e7)

Reply:
Why did you take back my code review after harassing me via Slack?